client/cli/src/commands/runcmd.rs

```diff
@@ -298,16 +275,16 @@ impl RunCmd {
 	{
 		self.shared_params.update_config(&mut config, spec_factory, version)?;
 
-		let password = if self.password_interactive {
+		let password = if self.shared_params.password_interactive {
 			#[cfg(not(target_os = "unknown"))]
 			{
 				Some(input_keystore_password()?.into())
 			}
 			#[cfg(target_os = "unknown")]
 			None
-		} else if let Some(ref file) = self.password_filename {
+		} else if let Some(ref file) = self.shared_params.password_filename {
 			Some(fs::read_to_string(file).map_err(|e| format!("{}", e))?.into())
-		} else if let Some(ref password) = self.password {
+		} else if let Some(ref password) = self.shared_params.password {
```

Comment:
If you moved some params to shared_params, you also need to move this code in the `update_config` of shared_params.

Reply:
honestly not even sure what `update_config` does or why it exists.

Reply:
sc-cli provides an API to build a CLI. This API will help you build a Configuration (sc-service) used to parametrize your blockchain. The current pattern of sc-cli is:

```rust
let mut config = Configuration::new();
cmd.init();
cmd.update_config(config);
cmd.run(config);
```

Where you first create a default sc_service::Configuration for your blockchain. Then init things specialized to the CLI itself (logging). Then update the sc_service::Configuration with the parameters of the CLI. Then run the node (or subcommand) using the configuration.

So `update_config` is always a method implemented on a structopt to update the Configuration with the parameters of that structopt.